src/pages/ProductListPage/FilterSection/FilterProduct.jsx

```diff
@@ -1,82 +1,263 @@
+//             return { categorytype : state.categorytype.filter((i)=>
+//                 state.categorytype.indexOf(i) !== state.categorytype.indexOf(action.categorytype))
+//             }
+//         }else{
+//             return { categorytype : state.categorytype.push(action.categorytype)}
+//         }
+//     }
+// }
+
+
 
 const FilterProduct = ({props}) =>{
     const {ProductList,setProductList} = props;
+
     console.log("filter product", ProductList,setProductList);
-  return (
-    <aside className="aside">
-        {/* <div className="toggler">
-            <span className="material-icons-round ">
-                cancel
-            </span>
-        </div> */}
-        <div className="aside-header">
-            <h3>Filters</h3>
-            <a>Clear</a>
-        </div>
-        <div className='aside-body'>
-            <div className="price-container">
-                <h4>Price</h4>
-                <input type="range" value="50" max="100" min="0" className="price-slider"/>
-            </div>
-            <div className="product-page-category">
-                <h4>Category</h4>
-                <div>
-                    <input className="checkbox-sneakers" type="checkbox" name="sneakers" value="sneakers"/>
-                    <label htmlFor="sneakers">Sneakers</label>
-                </div>
-                <div>
-                    <input className="checkbox-vans" type="checkbox" name="vans" value="vans"/>
-                    <label htmlFor="vans">vans</label>
-                </div>
-                <div>
-                    <input className="checkbox-converse" type="checkbox" name="converse" value="converse"/>
-                    <label htmlFor="converse">converse</label>
-                </div>
-                <div>
-                    <input className="checkbox-yezzy" type="checkbox" name="yezzy" value="yezzy"/>
-                    <label htmlFor="yezzy">yezzy</label>
-                </div>
+    // const [state,dispatch] = useReducer(FilterByInStock,{categorytype:[]})
+    const [categoryType,setCategoryType] = useState([]);
+    const [feature,setFeature] = useState([]);
+    const [sortByPrice,setSortByType] =useState(""); // hightolow = false & lowtohigh =true
+    const [rating,setRating] =useState(null);
+    const [productType,setProductType] =useState(null); // shoe = true & slides =false
+    const [productStockType,setProductStockType] =useState(null);  // topseller = true & newest =false
+    
+    useEffect(()=>{
+        // console.log("state useEffect",categoryType,feature,"sortByPrice",sortByPrice,"rating");
+        var obj ={
+            categoryType:categoryType,
+            feature:feature,
+            sortByPrice:sortByPrice,
+            rating:rating,
+            productType:productType,
+            productStockType:productStockType
+        }
+        console.log(obj);
+        Filter(obj)
+       
+    },[categoryType,feature,sortByPrice,rating,productType,productStockType])
```

Comment:
What is the use of using use-effect for filtering.

Reply:
this will update the product list once all filter section state are set so after every filter type is set state then we update the product list